fletcher/string_array.py

```diff
@@ -245,16 +246,52 @@ def _missing_capactiy(capacity):
     return int(math.ceil(capacity / 8))
 
 
-@pd.api.extensions.register_series_accessor("fr_text")
-@pd.api.extensions.register_series_accessor("text")
+@pd.api.extensions.register_series_accessor("fr_str")
+class TextAccessorExt:
+    def __init__(self, obj):
+        """Accessor for pandas exposed as ``.fr_str``.
+        fletcher functionality will be used if available otherwise dt functions are invoked."""
+        if not isinstance(obj.values, FletcherBaseArray):
+            # call StringMethods to validate the input obj
+            StringMethods(obj)
+        self.obj = obj
+
+    def __getattribute__(self, name):
```

Comment:
Can you factor this out into some well-named helper methods? The nesting makes this quite hard to understand.